src/rust/runtime/queue/qresult.rs

```diff
@@ -7,16 +7,33 @@
 
 use crate::runtime::{
     fail::Fail,
+    memory::DemiBuffer,
     network::types::Port16,
     QDesc,
 };
-use ::std::net::Ipv4Addr;
+use ::std::{
+    fmt,
+    net::{
+        Ipv4Addr,
+        SocketAddrV4,
+    },
+};
 
 //==============================================================================
-// Enumerations
+// Structures
 //==============================================================================
 
+pub enum OperationResult {
+    Connect,
+    Accept((QDesc, SocketAddrV4)),
+    Push,
+    // TODO: Drop wrapping Option.
+    Pop(Option<SocketAddrV4>, DemiBuffer),
+    Failed(Fail),
+}
+
 /// Result for IO Queue Operations
+#[deprecated]
```

Comment:
What is deprecated on this?

Reply:
It looks like we always just fill in the demi_qresult_t without using the Result, so I thought perhaps we could deprecate it. It looks like nothing in our examples or test directories depend on it.

Reply:
Agree! Could we have an issue for this?

Reply:
Sure, I took it out of this PR!